django/db/models/query.py

```diff
@@ -301,13 +302,19 @@ def __or__(self, other):
     # METHODS THAT DO DATABASE QUERIES #
     ####################################
 
-    def iterator(self):
+    def iterator(self, chunk_size=2000):
         """
         An iterator over the results from applying this QuerySet to the
         database.
         """
-        use_chunked_fetch = not connections[self.db].settings_dict.get('DISABLE_SERVER_SIDE_CURSORS')
-        return iter(self._iterable_class(self, chunked_fetch=use_chunked_fetch))
+        if chunk_size is None:
+            use_chunked_fetch = False
+            chunk_size = 2000
```

Comment:
Does this value have any significance or is it ignored if `use_chunked_fetch=False`? 

A question is what happens on non-PostgreSQL databases? The documentation should address that.